src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -514,18 +514,18 @@ private static boolean startsWithInheritDoc(DetailNode root) {
         boolean found = false;
         final DetailNode[] children = root.getChildren();
 
-        for (int i = 0; !found; i++) {
-            final DetailNode child = children[i];
-            if (child.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG
-                    && child.getChildren()[1].getType() == JavadocTokenTypes.INHERIT_DOC_LITERAL) {
-                found = true;
-            }
-            else if (child.getType() != JavadocTokenTypes.LEADING_ASTERISK
+        for (DetailNode child : children) {
+            if ((child.getType() == JavadocTokenTypes.TEXT
+                    || child.getType() == JavadocTokenTypes.HTML_ELEMENT)
```

Comment:
As you are not changing any parts of expressions, please restore their order to minimize git dif.

Reply:
done